pkg/processor/batchprocessor.go

```diff
@@ -101,13 +101,19 @@ func (s *segmentsBatch) poll() {
 				for i := 0; i < len(batch); i++ {
 					segIdStrs := segIdRegexp.FindStringSubmatch(*batch[i])
 					if len(segIdStrs) != 2 {
-						log.Debugf("Failed to match \"id\" in segment: ", *batch[i])
+						log.Debugf("Failed to match \"id\" in segment: %v", *batch[i])
 						continue
 					}
 					batchesMap[segIdStrs[1]] = *batch[i]
 				}
 				for _, unprocessedSegment := range r.UnprocessedTraceSegments {
 					telemetry.T.SegmentRejected(1)
+					// Print all segments since don't know which exact one is invalid.
+					if unprocessedSegment.Id == nil {
+						log.Debugf("Received invalid unprocessed segment id from X-Ray: %v", unprocessedSegment)
```

Comment:
Nitpick, X-Ray service? This is X-Ray daemon code so it is strange to say "from X-Ray". Also it is not invalid segment id but the service cannot parse out a id to return, so I'd suggest a clearer message here.

Reply:
update to `Received nil unprocessed segment id from X-Ray service`